lib/bn/bn128_arithmetic.ex

```diff
@@ -86,72 +110,113 @@ defmodule BN.BN128Arithmetic do
     end
   end
 
-  @spec double(Point.t()) :: Point.t()
-  defp double(point) do
-    if point.y.value == 0 do
-      {:ok, result} = Point.new(0, 0, modulus: point.modulus)
+  @spec double(point()) :: point()
+  def double({x, y} = {%FQ{}, %FQ{}}) do
+    if y.value == 0 do
+      {FQ.new(0), FQ.new(0)}
+    else
+      double_y = FQ.mult(y, 2)
+
+      lambda =
```

Comment:
Would it make sense to extract this `lambda` into its own function? Not sure if there's enough info to be standalone, so just a thought.